test/e2e/storage/vsphere_utils.go

```diff
@@ -53,40 +61,66 @@ func verifyVSphereDiskAttached(vsp *vsphere.VSphere, volumePath string, nodeName
 	return isAttached, err
 }
 
-// Wait until vsphere vmdk is deteched from the given node or time out after 5 minutes
-func waitForVSphereDiskToDetach(vsp *vsphere.VSphere, volumePath string, nodeName types.NodeName) error {
+// Wait until vsphere vmdk moves to expected state on the given node, or time out after 6 minutes
+func waitForVSphereDiskStatus(vsp *vsphere.VSphere, volumePath string, nodeName types.NodeName, expectedState volumeState) error {
 	var (
-		err            error
-		diskAttached   = true
-		detachTimeout  = 5 * time.Minute
-		detachPollTime = 10 * time.Second
+		err          error
+		diskAttached bool
+		currentState volumeState
+		timeout      = 6 * time.Minute
+		pollTime     = 10 * time.Second
 	)
-	if vsp == nil {
-		vsp, err = vsphere.GetVSphere()
-		if err != nil {
-			return err
-		}
-	}
-	err = wait.Poll(detachPollTime, detachTimeout, func() (bool, error) {
+	err = wait.Poll(pollTime, timeout, func() (bool, error) {
 		diskAttached, err = verifyVSphereDiskAttached(vsp, volumePath, nodeName)
 		if err != nil {
 			return true, err
 		}
-		if !diskAttached {
-			framework.Logf("Volume %q appears to have successfully detached from %q.",
-				volumePath, nodeName)
+
+		if diskAttached {
+			currentState = volumeStateAttached
+		} else {
+			currentState = volumeStateDetached
+		}
+
+		if currentState == expectedState {
+			if expectedState == volumeStateDetached {
+				framework.Logf("Volume %q has successfully detached from %q.", volumePath, nodeName)
```

Comment:
would be cleaner to see these msgs and the msgs below in a map indexed by state.  Eg:
`framework.Logf(successStateMsgs[expectedState])`

Reply:
A state map could simplify this and below. Eg:
```
var attachedState = map[bool]int{
            true: volumeStateAttached,
            false: volumeStateDetached,
}
var attachedStateMsgs = map[int]string{
            volumeStateAttached: "attached to",
            volumeStateDetached : "detached from",
}
...
// in wait.Poll func...
...
       currentState := attachedState[diskAttached]
       if currentState == expectedState {
             framework.Logf("Volume %q has successfully %s %q", volumePath, attachedStateMsgs[currentState], nodeName)
             return true, nil
       }
      framework.Logf("Waiting for Volume %q to %s %q.", volumePath, attachedStateMsgs[currentState], nodeName)
      return false, nil
})
```